src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Animal) || animal instanceof Carnivore || animal.hidden) {
+      return;
+    }
+
+    animal.health -= 50;
+
+    if (animal.health <= 0) {
+      const index = Animal.alive.indexOf(animal);
+      if (index !== -1) {
+        Animal.alive.splice(index, 1);
```

Comment:
The current use of `indexOf` and `splice` can lead to unintended behavior if the `index` is `-1`. Ensure that you check the `index` before calling `splice`. Alternatively, consider using the `filter` method to handle this more elegantly.